.pylintrc

```diff
@@ -60,7 +66,70 @@ confidence=
 # --enable=similarities". If you want to run only the classes checker, but have
 # no Warning level messages displayed, use"--disable=all --enable=classes
 # --disable=W"
-#disable=
+disable=
+    # These are disabled by default:
+    import-star-module-level,
+    old-octal-literal,
+    oct-method,
+    print-statement,
+    unpacking-in-except,
+    parameter-unpacking,
+    backtick,
+    old-raise-syntax,
+    old-ne-operator,
+    long-suffix,
+    dict-view-method,
+    dict-iter-method,
+    metaclass-assignment,
+    next-method-called,
+    raising-string,
+    indexing-exception,
+    raw_input-builtin,
+    long-builtin,
+    file-builtin,
+    execfile-builtin,
+    coerce-builtin,
+    cmp-builtin,
+    buffer-builtin,
+    basestring-builtin,
+    apply-builtin,
+    filter-builtin-not-iterating,
+    using-cmp-argument,
+    useless-suppression,
+    range-builtin-not-iterating,
+    suppressed-message,
+    no-absolute-import,
+    old-division,
+    cmp-method,
+    reload-builtin,
+    zip-builtin-not-iterating,
+    intern-builtin,
+    unichr-builtin,
+    reduce-builtin,
+    standarderror-builtin,
+    unicode-builtin,
+    xrange-builtin,
+    coerce-method,
+    delslice-method,
+    getslice-method,
+    setslice-method,
+    input-builtin,
+    round-builtin,
+    hex-method,
+    nonzero-method,
+    map-builtin-not-iterating,
+    # These are disabled in openshift-ansible:
+    # disabling rules is not to be reported as a violation
+    locally-disabled,
+    # some files are to be ignored
+    file-ignored,
+    # it is expected to have TODO notes
+    fixme,
+    # sometimes it's better to have no docstrings than placeholders or incorrect text
+    missing-docstring,
```

Comment:
> I think we should get more input from the larger group

:+1:

Yes, I agree!

---

While disabling was the quick-fix for getting a green run from pytest, I did fix some other issues like long lines, unused arguments.

For docstrings and names, instead of adding several inline disables, I used the reasoning I've documented.

## missing-docstrings

I've seen throughout the repo several docstrings that just state the obvious, apparently just to satisfy pylint, and I remember PRs fixing copypasta incorrect docstrings.

Looking only at files open in my editor:

- `""" Tests for the modify_yaml Ansible module. """`
- `"""A setuptools based setup module.\n\n"""`

I'd say those are not very useful.

I'm in favor of people giving good names to their functions, classes, methods, variables, and documenting what needs to be documented. If that's something we collectively care about, we need to be more attentive to that during code review.

I feel this pylint check is not enough and often forces us to add a `pylint: disable=missing-docstring` (64 occurrences in 45 files) or to add a non-sense docstring. That's just my opinion, I'm okay to keep it enabled if other people prefer it that way. In that case, we'd need to go fix the missing docstrings or add the inline disable comments.

## invalid-name

These are the invalid names when I remove `invalid-name` from the disabled list:

<details>
<summary>$ tox -e py35-ansible22-pylint</summary>

```
py35-ansible22-pylint installed: ansible==2.2.1.0,appdirs==1.4.2,astroid==1.4.9,cffi==1.9.1,coverage==4.3.4,cryptography==1.7.2,flake8==3.3.0,flake8-mutable==1.1.0,flake8-print==2.0.2,idna==2.2,isort==4.2.5,Jinja2==2.8.1,lazy-object-proxy==1.2.2,MarkupSafe==0.23,mccabe==0.6.1,mock==2.0.0,packaging==16.8,paramiko==2.1.2,pbr==1.10.0,py==1.4.32,pyasn1==0.2.3,pycodestyle==2.3.1,pycparser==2.17,pycrypto==2.6.1,pyflakes==1.5.0,pylint==1.6.5,pyOpenSSL==16.2.0,pyparsing==2.1.10,pytest==3.0.6,pytest-cov==2.4.0,PyYAML==3.12,setuptools-lint==0.5.2,six==1.10.0,wrapt==1.10.8,yamllint==1.6.1
py35-ansible22-pylint runtests: PYTHONHASHSEED='1093142714'
py35-ansible22-pylint runtests: commands[0] | sh -c git ls-files -co --exclude-standard '*.py' | grep -Ev '^(utils|roles/lib_(openshift|utils)/src)/' | xargs pylint
************* Module test_fakeopensslclasses
roles/openshift_certificate_expiry/test/test_fakeopensslclasses.py:14: [C0103(invalid-name), ] Invalid constant name "module_path"
roles/openshift_certificate_expiry/test/test_fakeopensslclasses.py:16: [C0103(invalid-name), ] Invalid constant name "openshift_cert_expiry"
roles/openshift_certificate_expiry/test/test_fakeopensslclasses.py:27: [C0103(invalid-name), TestFakeOpenSSLClasses.setUp] Invalid variable name "fp"
roles/openshift_certificate_expiry/test/test_fakeopensslclasses.py:32: [C0103(invalid-name), TestFakeOpenSSLClasses.setUp] Invalid variable name "fp"
roles/openshift_certificate_expiry/test/test_fakeopensslclasses.py:37: [C0103(invalid-name), TestFakeOpenSSLClasses.test_FakeOpenSSLCertificate_get_serial_number] Invalid method name "test_FakeOpenSSLCertificate_get_serial_number"
roles/openshift_certificate_expiry/test/test_fakeopensslclasses.py:41: [C0103(invalid-name), TestFakeOpenSSLClasses.test_FakeOpenSSLCertificate_get_notAfter] Invalid method name "test_FakeOpenSSLCertificate_get_notAfter"
roles/openshift_certificate_expiry/test/test_fakeopensslclasses.py:46: [C0103(invalid-name), TestFakeOpenSSLClasses.test_FakeOpenSSLCertificate_get_sans] Invalid method name "test_FakeOpenSSLCertificate_get_sans"
roles/openshift_certificate_expiry/test/test_fakeopensslclasses.py:61: [C0103(invalid-name), TestFakeOpenSSLClasses.test_FakeOpenSSLCertificate_get_sans_no_sans] Invalid method name "test_FakeOpenSSLCertificate_get_sans_no_sans"
roles/openshift_certificate_expiry/test/test_fakeopensslclasses.py:66: [C0103(invalid-name), TestFakeOpenSSLClasses.test_FakeOpenSSLCertificate_get_subject] Invalid method name "test_FakeOpenSSLCertificate_get_subject"
roles/openshift_certificate_expiry/test/test_fakeopensslclasses.py:76: [C0103(invalid-name), TestFakeOpenSSLClasses.test_FakeOpenSSLCertificate_get_subject_san_cert] Invalid method name "test_FakeOpenSSLCertificate_get_subject_san_cert"
************* Module openshift_master_facts_default_predicates_tests
roles/openshift_master_facts/test/openshift_master_facts_default_predicates_tests.py:163: [C0103(invalid-name), test_short_version_deployment_type_kwargs] Invalid function name "test_short_version_deployment_type_kwargs"
roles/openshift_master_facts/test/openshift_master_facts_default_predicates_tests.py:163: [C0103(invalid-name), test_short_version_deployment_type_kwargs] Invalid argument name "short_version_deployment_type_kwargs_fixture"
roles/openshift_master_facts/test/openshift_master_facts_default_predicates_tests.py:172: [C0103(invalid-name), short_version_deployment_type_kwargs_fixture] Invalid function name "short_version_deployment_type_kwargs_fixture"
************* Module openshift_master_facts_bad_input_tests
roles/openshift_master_facts/test/openshift_master_facts_bad_input_tests.py:33: [C0103(invalid-name), TestOpenShiftMasterFactsBadInput.test_missing_short_version_and_missing_openshift_release] Invalid method name "test_missing_short_version_and_missing_openshift_release"
************* Module openshift_master_facts_default_priorities_tests
roles/openshift_master_facts/test/openshift_master_facts_default_priorities_tests.py:151: [C0103(invalid-name), test_short_version_deployment_type_kwargs] Invalid function name "test_short_version_deployment_type_kwargs"
roles/openshift_master_facts/test/openshift_master_facts_default_priorities_tests.py:151: [C0103(invalid-name), test_short_version_deployment_type_kwargs] Invalid argument name "short_version_deployment_type_kwargs_fixture"
roles/openshift_master_facts/test/openshift_master_facts_default_priorities_tests.py:160: [C0103(invalid-name), short_version_deployment_type_kwargs_fixture] Invalid function name "short_version_deployment_type_kwargs_fixture"
ERROR: InvocationError: "/usr/bin/sh -c git ls-files -co --exclude-standard '*.py' | grep -Ev '^(utils|roles/lib_(openshift|utils)/src)/' | xargs pylint"
___________________________________________________________ summary ___________________________________________________________
ERROR:   py35-ansible22-pylint: commands failed
```
</details>


I don't see value renaming those things?

They'd add to a list of other 40 occurrences of `disable=invalid-name` in 33 files.